### PR TITLE
Fix timestamp parsing in ORC reader for timezones without transitions

### LINKED ISSUE #6947: [BUG] Loading timestamps from ORC sometimes returns incorrect results
**Describe the bug**
Some of our Spark integration tests for ORC have been failing with timestamp data mismatches.  The test writes out randomly generated data (with a deterministic RNG seed) using Spark CPU then reads the data with Spark CPU and then Spark using the RAPIDS Accelerator plugin.  Sometimes, not always, the tests will mismatch on timestamp data, e.g.:
```
03:58:29  cpu = datetime.datetime(1898, 5, 12, 21, 45, 8, 426000)
03:58:29  gpu = datetime.datetime(1898, 5, 13, 6, 47, 12, 426000)
```

**Steps/Code to reproduce bug**
This has been hard to reproduce outside of the Spark integration tests and even outside of the Jenkins pipelines for those tests.  Latest news is we believe it is triggered by using the GMT timezone on a CentOS 7 machine.  We've never seen it fail on an Ubuntu 18.04 machine with the GMT timezone.  @nvdbaranec has had some success trying to isolate the issue.

**Expected behavior**
ORC timestamps are loaded via cudf the same way they are loaded by Spark for the GMT timezone.

### PR DESCRIPTION
**Pull request by vuule**
Fixes #6947 

When TZif file has no transitions (e.g. GMT), `build_timezone_transition_table` has an out-of-bounds read that leads to undefined behavior and intermittent issues.

This PR makes two changes to behavior:
1. When there are no transitions, the ancient rule is initialized from the first time offset (instead of the first transition rule, which does not exist in this case).
2. When there are no transitions and the time offset is zero, an empty table is returned (avoid using a no-op table in CUDA).

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]
Pytest changes LGTM

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
Rerun tests

**Review by nvdbaranec — APPROVED** [review]

**Review by rgsl888prabhu — APPROVED** [review]

File: cpp/src/io/orc/timezone.cpp [hunk 1]
```diff
@@ -398,9 +398,22 @@ timezone_table build_timezone_transition_table(std::string const &timezone_name)
...
+    ttimes[0]  = ttimes[earliest_std_idx];
+    offsets[0] = offsets[earliest_std_idx];
+  } else {
+    if (tzf.typecnt() == 0 || tzf.ttype[0].utcoff == 0) {
+      // No transitions, offset is zero; Table would be a no-op.
+      // Return an empty table to speed up parsing.
+      return {};
+    }
+    // No transitions to use for the time/offset - use the first offset and apply to all timestamps
+    CUDF_EXPECTS(tzf.typecnt() != 0, "Need at least one offset");
```
Comment (davidwendt) [inline cpp/src/io/orc/timezone.cpp:413]: This check seems unnecessary since the if-statement above will catch `== 0`.

Comment (vuule) [reply cpp/src/io/orc/timezone.cpp:413]: you're right. Not sure what I was thinking.

**Review by davidwendt — APPROVED** [review]

**Pull request merged**